data_safe_haven/infrastructure/components/dynamic/file_upload.py

```diff
@@ -117,6 +117,10 @@ def diff(
             )
         return self.partial_diff(old_props, new_props, [])
 
+    def refresh(self, props: dict[str, Any]) -> dict[str, Any]:
+        """TODO: check whether file has changed"""
```

Comment:
Same as above

Reply:
As above - I think it shouldn't be too hard to do, but isn't the focus of this PR.